pinot-core/src/main/java/org/apache/pinot/core/realtime/impl/dictionary/BytesOnHeapMutableDictionary.java

```diff
@@ -20,20 +20,29 @@
 
 import java.util.Arrays;
 import javax.annotation.Nonnull;
+import javax.xml.bind.DatatypeConverter;
 import org.apache.pinot.common.utils.primitive.ByteArray;
 
 
 /**
  * OnHeap mutable dictionary of Bytes type.
  */
 public class BytesOnHeapMutableDictionary extends BaseOnHeapMutableDictionary {
+
   private ByteArray _min = null;
   private ByteArray _max = null;
 
   @Override
   public int indexOf(Object rawValue) {
     assert rawValue instanceof byte[];
-    return getDictId(new ByteArray((byte[]) rawValue));
+    byte[] bytes;
+    // Convert hex string to byte[].
+    if (rawValue instanceof String) {
+      bytes = DatatypeConverter.parseHexBinary((String) rawValue);
```

Comment:
Other parts of the code use org.apache.commons.codec.binary.Hex for encoding/decoding byte-array to String. Either one is fine, but all code should use the same encoding/decoding.

Reply:
sounds good, updated with org.apache.commons.codec.binary.Hex api.

Reply:
sounds good, updated with org.apache.commons.codec.binary.Hex api.